### PR TITLE
Remove saga from the content manager

### PR DESCRIPTION
**Pull request by soupette**
#### Description of what you did:
This PR removes `redux/saga` from the content manager plugin.

#### My PR is a:

- [ ] 💥 Breaking change
- [ ] 🐛 Bug fix
- [X] 💅 Enhancement
- [ ] 🚀 New feature

#### Main update on the:

- [ ] Admin
- [ ] Documentation
- [ ] Framework
- [X] Plugin

#### Manual testing done on the following databases:

- [X] Not applicable
- [ ] MongoDB
- [ ] MySQL
- [ ] Postgres
- [ ] SQLite

### CODE REVIEW

File: packages/strapi-plugin-content-manager/admin/src/containers/ListView/utils/generateSearchFromObject.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      }
+
+      return acc;
+    }, [])
+    .join('&');
+
+  return search;
+};
+
+export default generateSearchFromObject;
```
Comment (HichamELBSI) [inline packages/strapi-plugin-content-manager/admin/src/containers/ListView/utils/generateSearchFromObject.js]: Maybe you can add some test for this function.

File: packages/strapi-plugin-content-manager/admin/src/containers/Main/utils/createPossibleMainFieldsForModelsAndComponents.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      ].includes(get(attributes, [attr, 'type'], ''));
+    });
+
+    acc[current.uid] = possibleMainFields;
+
+    return acc;
+  }, {});
+};
+
+export default createPossibleMainFieldsForModelsAndComponents;
```
Comment (HichamELBSI) [inline packages/strapi-plugin-content-manager/admin/src/containers/Main/utils/createPossibleMainFieldsForModelsAndComponents.js]: Some test for this function too.

Comment (soupette) [reply packages/strapi-plugin-content-manager/admin/src/containers/Main/utils/createPossibleMainFieldsForModelsAndComponents.js]: The tests will come in another PR after we are finished removing the saga dependency.

File: packages/strapi-plugin-content-manager/admin/src/containers/ListView/utils/generateSearchFromObject.js (hunk 1 above)
Comment (soupette) [reply packages/strapi-plugin-content-manager/admin/src/containers/ListView/utils/generateSearchFromObject.js]: Sure I agree but they will come with another PR after we are finished removing redux-saga.

Comment (soupette) [reply packages/strapi-plugin-content-manager/admin/src/containers/ListView/utils/generateSearchFromObject.js]: This is some old code that I have put in a util to prepare the upcoming tests...

File: packages/strapi-plugin-content-manager/admin/src/containers/Main/utils/createPossibleMainFieldsForModelsAndComponents.js (hunk 2 above)
Comment (soupette) [reply packages/strapi-plugin-content-manager/admin/src/containers/Main/utils/createPossibleMainFieldsForModelsAndComponents.js]: Test have been added.

File: packages/strapi-plugin-content-manager/admin/src/containers/ListView/utils/generateSearchFromObject.js (hunk 1 above)
Comment (soupette) [reply packages/strapi-plugin-content-manager/admin/src/containers/ListView/utils/generateSearchFromObject.js]: Tests have been added.

**Comment by HichamELBSI** [comment]
LGTM

**Pull request merged**